lib/hubspot/deal.rb

```diff
@@ -68,12 +68,28 @@ def recent(opts = {})
       # @param company [Hubspot::Company] the company
       # @return [Array] Array of Hubspot::Deal records
       def find_by_company(company)
+        find_by_association company
+      end
+
+      # Find all deals associated to a contact
+      # {http://developers.hubspot.com/docs/methods/deals/get-associated-deals}
+      # @param contact [Hubspot::Contact] the contact
+      # @return [Array] Array of Hubspot::Deal records
+      def find_by_contact(contact)
+        find_by_association contact
+      end
+
+      # Find all deals associated to a contact or company
+      # {http://developers.hubspot.com/docs/methods/deals/get-associated-deals}
+      # @param object [Hubspot::Contact || Hubspot::Company] a contact or company
+      # @return [Array] Array of Hubspot::Deal records
+      def find_by_association(object)
         path = ASSOCIATED_DEAL_PATH
-        params = { objectType: :company, objectId: company.vid }
+        type = object.class.to_s.gsub('Hubspot::', '').downcase.to_sym
```

Comment:
We should verify that the object being passed in is something we and Hubspot support. Passing in a String would send an invalid request to Hubspot and returns error `Invalid enum constant 'string' for class ObjectType`. It would be nicer to return Hubspot::InvalidParams here if the object isn't of a type that is supported. Something like this would be ideal:
```
type = case object
       when Hubspot::Company then :company
       when Hubspot::Contact then :contact
       else raise Hubspot::InvalidParams, 'Instance type not supported'
       end
```

Reply:
Sorry about the delay. I was really busy. I am going to do these changes this week. Thanks for your reply.